code/clue.js

```diff
@@ -1,59 +1,454 @@
+    toggleAnimation(loaderId);
+    get('info').innerHTML = '';
+    setTimeout(function () {
+      toggleLoader(loaderId);
+      mystery.room = increaseChance(mystery.weapon);
+      style('roomCard').background = mystery.killer.color;
+      get('roomName').innerHTML = `${capitalizeFirstLetter(mystery.room.name)}`;
+      roomPicked = true;
+      style('revealButton').visibility = 'visible';
+    }, 1500);
+  } else {
+    showInfo(loaderId);
+  }
+};
+
+/* The function is called when the 'Reveal the Crime' button is clicked. Checks
+ if all cards have been picked and then toggles window to display a message 
+ with the mystery object  */
+const revealMystery = (windowId) => {
+  if (killerPicked && weaponPicked && roomPicked) {
+    toggleWindow(windowId);
+    get(
+      'mystery'
+    ).innerHTML = `The murder was commited by ${mystery.killer.firstName} 
+    ${mystery.killer.lastName} in the ${mystery.room.name} with a ${mystery.weapon.name}`;
+    style('mysteryCard').background = mystery.killer.color;
+    get(
+      'mysteryKiller'
+    ).innerHTML = `Mystery Killer: ${mystery.killer.firstName} 
+    ${mystery.killer.lastName}`;
+    get('mysteryKillerImage').src = mystery.killer.image;
+    get('mysteryRoom').innerHTML = `Room: ${capitalizeFirstLetter(
+      mystery.room.name
+    )}`;
+    get('mysteryWeapon').innerHTML = `Weapon: ${capitalizeFirstLetter(
+      mystery.weapon.name
+    )}`;
+  }
+};
+
+const restartGame = () => {
+  location.reload();
+};
+
+/* Function to randomly select one item from the array passed as a paramater */
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
+
+/* The function is called when the Let's play button is clicked to shuffle
+ favourite weapon and favourite room */
+const shuffleFavourite = (mysteryObject, favouriteObject) => {
+  mysteryObject.forEach((object) => {
+    object.favourite = randomSelector(favouriteObject).id;
+  });
+};
+
+/* Toggles the loader animation */
+const toggleAnimation = (loaderId) => {
+  get(loaderId).classList.toggle('loader-container-animation');
+};
+
+/* Hides the loader image when a card is picked */
+const toggleLoader = (loaderId) => {
+  style(loaderId).display = 'none';
+};
+
+/* Hides the current window and displays the next */
+const toggleWindow = (windowId) => {
+  const windowElement = get(windowId);
+  toggleWindowClass(windowId);
+  toggleWindowClass(windowElement.nextElementSibling.id);
+};
+
+/* Toggles between classes to hide/display a window */
+const toggleWindowClass = (id) => {
+  get(id).classList.toggle('visible');
+  get(id).classList.toggle('invisible');
+};
+
+/* Hides the reveal the crime button when the game is started */
+const hideButton = () => {
+  const button = (style('revealButton').visibility = 'hidden');
+};
+
+/* Increases the chance of the referred obeject's favourite property to be picked */
+const increaseChance = (object) => {
+  if (object === mystery.killer) {
```

Comment:
We usually store an `id` property in objects to compare those values instead of the whole objects